src/net-questdb-client/Ingress/Senders/HttpSender.cs

```diff
@@ -0,0 +1,394 @@
+    {
+        switch (code)
+        {
+            case HttpStatusCode.InternalServerError:
+            case HttpStatusCode.ServiceUnavailable:
+            case HttpStatusCode.GatewayTimeout:
+            case HttpStatusCode.InsufficientStorage:
+            case (HttpStatusCode)509: // Bandwidth Limit Exceeded
+            case (HttpStatusCode)523: // Origin Is Unreachable
+            case (HttpStatusCode)524: // A Timeout Occurred
+            case (HttpStatusCode)529: // Site is overloaded
+            case (HttpStatusCode)599: // Network Timeout Error
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    /// <summary>
+    ///     Applies retry logic (if required).
+    /// </summary>
+    /// <remarks>
+    ///     Until <see cref="QuestDBOptions.retry_timeout" /> has elapsed, the request will be retried
+    ///     with a small jitter.
+    /// </remarks>
+    /// <param name="response">The response triggering the retry.</param>
+    /// <param name="cts">The cancellation token source.</param>
+    /// <param name="retryIntervalMs">The base interval between retries.</param>
+    /// <returns></returns>
+    /// <exception cref="IngressError"></exception>
+    private async Task FinishOrRetryAsync(HttpResponseMessage response,
+        CancellationTokenSource? cts = default, int retryIntervalMs = 10)
+    {
+        var lastResponse = response;
+
+        if (!response.IsSuccessStatusCode)
+        {
+            if (!(IsRetriableError(response.StatusCode) && Options.retry_timeout > TimeSpan.Zero))
+            {
+                throw new IngressError(ErrorCode.ServerFlushError, response.ReasonPhrase);
+            }
+
+            var retryTimer = new Stopwatch();
+            retryTimer.Start();
+
+            var retryInterval = TimeSpan.FromMilliseconds(retryIntervalMs);
+
+            while (retryTimer.Elapsed < Options.retry_timeout)
+            {
+                var jitter = TimeSpan.FromMilliseconds(Random.Shared.Next(0, retryIntervalMs) - retryIntervalMs / 2.0);
+                await Task.Delay(retryInterval + jitter);
+
+                var (nextRequest, nextToken) = GenerateRequest();
+                lastResponse = await _client!.SendAsync(nextRequest, nextToken!.Token);
+                if (!lastResponse.IsSuccessStatusCode)
+                {
+                    if (IsRetriableError(lastResponse.StatusCode) && Options.retry_timeout > TimeSpan.Zero)
+                    {
+                        cts!.Dispose();
+                    }
+                }
+            }
+        }
+
+        if (!lastResponse.IsSuccessStatusCode)
+        {
+            throw new IngressError(ErrorCode.ServerFlushError, lastResponse.ReasonPhrase);
+        }
+
+        LastFlush = (lastResponse.Headers.Date ?? DateTimeOffset.UtcNow).UtcDateTime;
+        cts!.Dispose();
+        Buffer.Clear();
+    }
+
+    
+    public void Dispose()
+    {
+        _client.DisposeNullable();
```

Comment:
Java client sends data when closed, 


```Java
package com.example.sender;

import io.questdb.client.Sender;

public class HttpExample {
    public static void main(String[] args) {
        try (Sender sender = Sender.fromConfig("http::addr=localhost:9000;")) {
            sender.table("weather_sensor")
                    .symbol("id", "toronto1")
                    .doubleColumn("temperature", 23.5)
                    .doubleColumn("humidity", 0.49)
                    .atNow();
            sender.table("weather_sensor")
                    .symbol("id", "dubai2")
                    .doubleColumn("temperature", 41.2)
                    .doubleColumn("humidity", 0.34)
                    .atNow();
        }
    }
}
```

https://questdb.io/docs/clients/java_ilp/

this should have similar sematics